diff --git a/include/ariane_pkg.sv b/include/ariane_pkg.sv
index adc497e..f90c12e 100644
--- a/include/ariane_pkg.sv
+++ b/include/ariane_pkg.sv
@@ -457,14 +457,14 @@ package ariane_pkg;
     localparam int unsigned DCACHE_TAG_WIDTH   = riscv::PLEN - DCACHE_INDEX_WIDTH;
 `else
     // I$
-    localparam int unsigned CONFIG_L1I_SIZE    = 16*1024;
+    localparam int unsigned CONFIG_L1I_SIZE    = 8*1024;
     localparam int unsigned ICACHE_SET_ASSOC   = 4; // Must be between 4 to 64
     localparam int unsigned ICACHE_INDEX_WIDTH = $clog2(CONFIG_L1I_SIZE / ICACHE_SET_ASSOC);  // in bit, contains also offset width
     localparam int unsigned ICACHE_TAG_WIDTH   = riscv::PLEN-ICACHE_INDEX_WIDTH;  // in bit
     localparam int unsigned ICACHE_LINE_WIDTH  = 512; // in bit
     // D$
-    localparam int unsigned CONFIG_L1D_SIZE    = 32*1024;
-    localparam int unsigned DCACHE_SET_ASSOC   = 8; // Must be between 4 to 64
+    localparam int unsigned CONFIG_L1D_SIZE    = 16*1024;
+    localparam int unsigned DCACHE_SET_ASSOC   = 4; // Must be between 4 to 64
     localparam int unsigned DCACHE_INDEX_WIDTH = $clog2(CONFIG_L1D_SIZE / DCACHE_SET_ASSOC);  // in bit, contains also offset width
     localparam int unsigned DCACHE_TAG_WIDTH   = riscv::PLEN-DCACHE_INDEX_WIDTH;  // in bit
     localparam int unsigned DCACHE_LINE_WIDTH  = 512; // in bit
